### PR TITLE
solana-ibc: introduce ClientStateUpdate event

### PR DESCRIPTION
**Pull request by dhruvja**
Introduce ClientStateUpdate event which is emitted when client state
is updated and includes the updated serialised client state.

### CODE REVIEW

File: local.json
Comment (mina86) [inline local.json:1]: What’s this file?

File: solana/solana-ibc/programs/solana-ibc/src/execution_context.rs
Comment (mina86) [inline solana/solana-ibc/programs/solana-ibc/src/execution_context.rs:314]: Perhaps it’d be better to turn this into an Event? We could do the same with consensus state.

**Review by mina86 — APPROVED** [review]

File: solana/solana-ibc/programs/solana-ibc/src/execution_context.rs
Comment (dhruvja) [reply solana/solana-ibc/programs/solana-ibc/src/execution_context.rs:314]: Yes sounds good.

File: local.json
Comment (dhruvja) [reply local.json:1]: oh sorry, its the keypair of the buffer i was using to deploy.

**Comment by dhruvja** [comment]
@mina86 Lets close this in favor of #316

**Comment by mina86** [comment]
Or just merge this one once I merge mine into this one ;)

**Comment by dhruvja** [comment]
Sounds good.

**Pull request merged**
